src/components/Input/BoolInput.tsx

```diff
@@ -1,35 +1,76 @@
 import { JSX } from 'react';
-import { Checkbox, Group } from '@mantine/core';
+import { Checkbox, Group, MantineStyleProps } from '@mantine/core';
 
-import { Label } from '../Label/Label';
+import { InfoBox } from '@/components/InfoBox/InfoBox';
+import { RequireAtLeastOne } from '@/types/types';
 
-interface Props {
-  name: string;
-  description: string | JSX.Element;
-  value: boolean;
-  setValue: (value: boolean) => void;
-  label?: JSX.Element;
+interface BaseProps extends MantineStyleProps {
+  onChange?: (value: boolean) => void;
+  info?: string | JSX.Element;
   disabled?: boolean;
 }
 
+// Either the label or the ariaLabel must be provided, or both
+interface Label {
+  label?: string;
+  ariaLabel?: string;
+}
+type Labels = RequireAtLeastOne<Label, 'label' | 'ariaLabel'>;
+
+// If value is provided, defaultChecked must not be provided
+interface PropsWithValue extends BaseProps {
+  value: boolean;
+  defaultChecked?: never;
+}
+
+// If defaultChecked is provided, value must not be provided, but it's also possible to
+// not specify either, in which case defaultChecked will be considered false by default
+interface PropsWithDefault extends BaseProps {
+  defaultChecked?: boolean;
+  value?: never;
+}
+
+type Props = (PropsWithValue | PropsWithDefault) & Labels;
```

Comment:
What's the reason for not extending CheckboxProps? I'm not sure I understand the reasoning behind handling PropsWithDefault and PropsWithValue like we do above here..?

Reply:
iirc mantine allows both value and defaultchecked?

Reply:
Sure a todo is fine